### ISSUE TITLE
Incorrect glass rendering on Halo: CE

### ISSUE THREAD

**Issue by Vaporeon** [iXit/wine-nine-standalone#98]
This is likely due to the game doing undefined behaviour as the shader uses a mismatched image and sampler (iirc sampler is a cubemap but the image is a 2D texture)

Working:
![Screenshot_20210303_234553_working](https://user-images.githubusercontent.com/12814075/109794851-2811aa00-7c7b-11eb-9b8f-5956d43d7871.png)

Gallium Nine:
![Screenshot_20210303_234233_gallium_nine](https://user-images.githubusercontent.com/12814075/109794899-38c22000-7c7b-11eb-81ed-50c9934706d4.png)

**Comment by Vaporeon** [comment]
Unrelated but gallium nine has taken a massive performance hit in this game on recent WINE versions. I don't think the regression is in nine itself, so I'm not sure if it's worth opening an issue about it.

**Comment by dungeon007** [comment]
"Unrelated but gallium nine has taken a massive performance hit in this game on recent WINE versions."
Open it, that could be interesting, if you know on which WINE version that massive performance hit started.

**Comment by dungeon007** [comment]
Seems like that glass shader is working with ForceShader=14 in config.txt or -use14 switch, -use11 draws it too.
 At least on Radeon hardware that seems working, just not with default again so -use20. Neither wined3d draws that as i see 🤣

**Comment by Vaporeon** [comment]
wine3d does not account for this case, neither do modern Windows drivers.
Using a lower pixel shader version is also not entirely working as it uses a different shader path that breaks numerous other things.

The Glass only renders correctly on ~2003 ATI drivers, or on modern cards using dgVoodo2 or DXVK.

**Comment by dungeon007** [comment]
On the other hand dxvk seems to draw that via either radv or amdvlk with -use20 too, so yeah a bug.

**Comment by dungeon007** [comment]
That is something really BAD, if it does not work on any driver 🤣

**Comment by Vaporeon** [comment]
Yeah DXVK already has a specific check to handle this exact issue in this game, so it works on there.

**Comment by axeldavy** [comment]
I guess at some point we'll need to implement the same kind of game hacks DXVK does.

**Comment by Vaporeon** [comment]
Yeah, unfortunately.
I friend of mine tried to find a fix for this client side so it could be done in Chimera, but could not find the right thing to patch.
